packages/sdk/src/messaging/ack_manager.ts

```diff
@@ -0,0 +1,166 @@
+import { createDecoder } from "@waku/core";
+import {
+  IDecodedMessage,
+  IDecoder,
+  IFilter,
+  IStore,
+  NetworkConfig
+} from "@waku/interfaces";
+import { createRoutingInfo } from "@waku/utils";
+
+import { MessageStore } from "./message_store.js";
+import { IAckManager } from "./utils.js";
+
+type AckManagerConstructorParams = {
+  messageStore: MessageStore;
+  filter: IFilter;
+  store: IStore;
+  networkConfig: NetworkConfig;
+};
+
+export class AckManager implements IAckManager {
+  private readonly messageStore: MessageStore;
+  private readonly filterAckManager: FilterAckManager;
+  private readonly storeAckManager: StoreAckManager;
+  private readonly networkConfig: NetworkConfig;
+
+  private readonly subscribedContentTopics: Set<string> = new Set();
+
+  public constructor(params: AckManagerConstructorParams) {
+    this.messageStore = params.messageStore;
+    this.networkConfig = params.networkConfig;
+
+    this.filterAckManager = new FilterAckManager(
+      this.messageStore,
+      params.filter
+    );
+
+    this.storeAckManager = new StoreAckManager(this.messageStore, params.store);
+  }
+
+  public start(): void {
+    this.filterAckManager.start();
+    this.storeAckManager.start();
+  }
+
+  public async stop(): Promise<void> {
+    await this.filterAckManager.stop();
+    this.storeAckManager.stop();
+    this.subscribedContentTopics.clear();
+  }
+
+  public async subscribe(contentTopic: string): Promise<boolean> {
+    if (this.subscribedContentTopics.has(contentTopic)) {
+      return true;
+    }
+
+    this.subscribedContentTopics.add(contentTopic);
```

Comment:
multiple calls to `subscribe` with the same content topic may trigger the early return condition prematurely